department-of-veterans-affairs/caseflow
# MattT/APPEALS-33-Fixes-Branch

Resolves [APPEALS-12679](https://vajira.max.gov/browse/APPEALS-12679) and [APPEALS-12680](https://vajira.max.gov/browse/APPEALS-12680)

### Description

- Hides the pre-docket banner on the Add/Remove Issues page in Intake if a adds a VHA request issue and opts out of pre-docketing the appeal.
- Routes the appeal to the appropriate tasks factory ([InititalTasksFactory](https://github.com/department-of-veterans-affairs/caseflow/blob/d9c588df05f1dd1fdd4d71187daa466cbbe427d1/app/workflows/initial_tasks_factory.rb#L1)) if the intake user has added a VHA request issue to an appeal but has opted out of pre-docketing it.
- Prevents the creation of VeteranRecordRequest tasks to be assigned to the VHA business line regardless of if the VHA issues have been sent to pre-docket. This is because VHA stakeholders have expressed that they do not utilize this task.

### Acceptance Criteria
- [ ] Pre-docket banner does not appear on Add/Remove Issues page if none of the request issues on the appeal are elected for pre-docket.
- [ ] Appeals are not routed to pre-docket if all issues have been opted out of the pre-docket workflow, even if some of them are VHA issues.
- [ ] VeteranRecordRequest tasks are never assigned to the VHA business line.

### Testing Plan
1. Log into Caseflow as an Intake user (ex: `BVAISHAW`)
2. Navigate to the Intake homepage ([local link](http://localhost:3000/intake))
3. Select "Decision Review Request: Board Appeal" from the "Which form are you processing?" dropdown.
4. Go to the next page and enter a veteran file number. Use `Veteran.first.file_number` in the Rails console to snag one if you need it.
5. Enter whatever you'd like on the Review page for the various inputs. Go to the next page.
6. On the Add/Remove Issues page, click on "Add issue".
7. Add a VHA issue and select 'Yes' for the pre-docket prompt.
8. Ensure that the blue pre-docket banner appears.
9. Delete the issue from step 7.
10. Add another VHA issue and select 'No' for the pre-docket prompt.
11. Ensure that the blue pre-docket banner does not appear.
12. Establish the appeal.
13. Make sure that the text "Appeal created and sent to VHA for document assessment." is **absent**, and that the text "Appeal created:" is **present**.
14. Click on the "Correct the issues" link, and then grab the appeal UUID from the URL.
15. Open the Rails console, and run the following commands:
```ruby
a = Appeal.find_by_uuid(<appeal-uuid-from-url>)

a.treee
=>                                     ┌────────────────────────────────────────────────────────────────┐
Appeal 2375 (D 221127-2375 Original)  │ ID    │ STATUS   │ ASGN_BY │ ASGN_TO │ UPDATED_AT              │
└── RootTask                          │ 10769 │ on_hold  │         │ Bva     │ 2022-12-21 15:35:37 UTC │
    └── DistributionTask              │ 10770 │ assigned │         │ Bva     │ 2022-12-21 15:35:37 UTC │
                                      └────────────────────────────────────────────────────────────────┘
```
Make sure there is not a `PreDocketTask` or `VeteranRecordRequest` task in the task tree.

16. Repeat an intake and add a VHA pre-docket issue to the new appeal.
17. Take a look at that task tree in the Rails console, largely repeating step 15:
```ruby
a = Appeal.find_by_uuid(<appeal-uuid-from-url>)

a.treee
=>                               ┌─────────────────────────────────────────────────────────────────────────────┐
Appeal 2373 (D 221128-2373 Original)  │ ID    │ STATUS   │ ASGN_BY  │ ASGN_TO             │ UPDATED_AT              │
└── RootTask                          │ 10763 │ on_hold  │          │ Bva                 │ 2022-12-21 14:00:13 UTC │
    ├── PreDocketTask                 │ 10764 │ on_hold  │ BVAISHAW │ BvaIntake           │ 2022-12-21 14:00:13 UTC │
        └── VhaDocumentSearchTask     │ 10765 │ assigned │ BVAISHAW │ VhaCamo             │ 2022-12-21 14:00:13 UTC │
                                      └─────────────────────────────────────────────────────────────────────────────┘
```
There should be a `PreDocketTask` in this tree, but there should still not be a `VeteranRecordRequest` task.
